### PR TITLE
Reducing compiler directives (NETSTANDARD)

### PR DESCRIPTION
**Pull request by T-Gro**
Eliminate usage of NETSTANDARD compiler directives for main libraries where we have target frameworks netstandard(20/21) only => NETSTANDARD was defined in both cases, so the compiler always took only 1 branch of the #if .
- Fsharp.Compiler.Service
- Fsharp.Core

Reduce usage of NETSTANDARD conditionals in tests where needed.

Encapsulate #if - then #else pattern for tests that have been:
- Either NetCoreApp or DESKTOP, never both
- A xUnit Fact before (This PR does not enforce moving from nunit to xunit for tests where this did not apply)
- The test project already referenced Test.Utilities before (a new dependency not added)

### CODE REVIEW

File: src/Compiler/Driver/CompilerConfig.fs [hunk 1]
```diff
@@ -1131,13 +1131,7 @@ type TcConfig private (data: TcConfigBuilder, validate: bool) =
                         yield clrFacades
 
                 | None ->
-                    // "there is no really good notion of runtime directory on .NETCore"
-#if NETSTANDARD
                     let runtimeRoot = Path.GetDirectoryName(typeof<Object>.Assembly.Location)
```
Comment (KevinRansom) [inline src/Compiler/Driver/CompilerConfig.fs:1134]: System.Runtime.InteropServices.RuntimeEnvironment.GetRuntimeDirectory() exists on netstandard 2.0.  It is probably a better solution than location of assembly containing int.

**Review by KevinRansom — DISMISSED** [review]
Looks good.

File: src/Compiler/Driver/CompilerConfig.fs (hunk 1 above)
Comment (T-Gro) [inline src/Compiler/Driver/CompilerConfig.fs:1134]:
```suggestion
                    let runtimeRoot = System.Runtime.InteropServices.RuntimeEnvironment.GetRuntimeDirectory()
```

**Review by 0101 — DISMISSED** [review]

**Comment by T-Gro** [comment]
@KevinRansom :
I have to rollback your proposal.

When changing runtimeRoot (from typeof<Object>.Assembly.Location  to GetRuntimeDirectory() ), a test for **FSI** /usesdkrefs**-** starts failing with:

Error creating evaluation session: System.AggregateException: One or more errors occurred. (Assembly: The system type 'System.Array' was required but **no referenced system DLL contained this type**, full path: System.Array)

**Comment by T-Gro** [comment]
> @KevinRansom : I have to rollback your proposal.
> 
> When changing runtimeRoot (from typeof.Assembly.Location to GetRuntimeDirectory() ), a test for **FSI** /usesdkrefs**-** starts failing with:
> 
> Error creating evaluation session: System.AggregateException: One or more errors occurred. (Assembly: The system type 'System.Array' was required but **no referenced system DLL contained this type**, full path: System.Array)

@KevinRansom : Funnily, there was a bug related to trimming separators which only revealed itself after doing this change.
Should be able to fix it and keep your proposal :)

**Review by 0101 — APPROVED** [review]

**Review by 0101 — APPROVED** [review]

**Review by psfinaki — APPROVED** [review]

**Review by KevinRansom — APPROVED** [review]

**Pull request merged**